articles/setting-up-cicd-for-python-packages-using-github-actions/index.md

```diff
@@ -0,0 +1,279 @@
+
+Continuous Integration is a set of practices that enable development teams to integrate code into version control repositories regularly. According to Atlass*i*an, *It’s a primary DevOps best practice, allowing developers to frequently merge code changes into a central repository where builds and tests then run.*
+
+### Continuous Delivery
+
+Continuous delivery is a software development practice where release to production is done automatically after a code changes.
+
+### Continuous Deployment
+
+Continuous Deployment (CD) is a software release process that uses automated testing to validate if changes to a codebase are correct and stable for immediate autonomous deployment to a production environment.
+
+This [Amazon](https://aws.amazon.com/devops/continuous-delivery/) blog explains in details the difference between these three.
+
+## Building a Python Package
+
+We will attempt to build a basic Python package that tells a user the time in another time zone and works on the command line. The code for this project is in [this repository](https://github.com/edeediong/timezone_checker).
+
+### Application Logic
+
+To build our basic python package, we need two components, namely a `[setup.py](http://setup.py)` file and an `src` folder containing our package logic. Let's start with the `src` folder: it should contain three files, namely `__init__.py`, `[logic.py](http://logic.py)` and `main.py`. 
+
+Copy and paste the following code snippet into `logic.py`.
+
+```python
+from datetime import datetime
+import pytz
+
+def area(location):
+    location = format_location(location)
+    for areas in pytz.all_timezones:
+        if location.lower() in areas.lower():
+            location = areas
+            tz = pytz.timezone(location)
+            date_now = datetime.now(tz)
+            formatted_date = date_now.strftime("%B %d, %Y %H:%M:%S")
+            print(f"{location} time: ", formatted_date)
+            break
+
+    else:
+        print("This location isn't on the tz database on Wikipedia")
+
+def area_zone(zone):
+    try:
+        zone = timezones(zone)
+        tz = pytz.timezone(zone)
+        date_now = datetime.now(tz)
+        formatted_date = date_now.strftime("%B %d, %Y %H:%M:%S")
+        print(f"{zone} time: ", formatted_date)
+
+    except Exception:
+        print("Timezone is not on the list. Consider using location instead.")
+
+def timezones(zone):
+	"""This function is used to handle situations of Daylight Saving Time that the standard library can't recognize."""
+    zones = {
+        "PDT": "PST8PDT",
+        "MDT": "MST7MDT",
+        "EDT": "EST5EDT",
+        "CDT": "CST6CDT",
+        "WAT": "Etc/GMT+1",
+        "ACT": "Australia/ACT",
+        "AST": "Atlantic/Bermuda",
+        "CAT": "Africa/Johannesburg",
+    }
+
+    try:
+        zones[zone]
+			
+    except:
+        return zone
+    return zones[zone]
+
+def format_location(location):
+    location = location.replace(" ", "_")
+    return location
+```
```

Comment:
If possible, try adding comments. That would help beginners.

Reply:
Hi @srishilesh , That's why I ensured all functions come with docstrings as these are comments themselves. I also explained these functions after the code themswelves.